malcolm/modules/scanning/parts/multipletriggerpart.py

```diff
@@ -0,0 +1,90 @@
+from typing import Sequence, Union
+
+from annotypes import Anno, Array, add_call_types
+
+from malcolm.core import CAMEL_RE, APartName, PartRegistrar
+from malcolm.modules import builtin, scanning
+
+# Pull re-used annotypes into our namespace in case we are subclassed
+APartName = APartName
+AMri = builtin.parts.AMri
+AInitialVisibility = builtin.parts.AInitialVisibility
+
+with Anno("Single or list of valid multiples"):
+    AMultiples = Union[Array[int]]
+
+
+class MultipleTriggerPart(builtin.parts.ChildPart):
+    """Part for informing a parent that it can have multiple
+    triggers.
+    """
+
+    def __init__(
+        self,
+        name: APartName,
+        mri: AMri,
+        valid_multiples: Union[AMultiples, Sequence[int], int] = None,
```

Comment:
When would we need anything other than 2 in this list?

Reply:
Good question! I thought potentially with 3 beam positions on K11 I'd keep it generic enough to be able to change it in the future.

In hindsight I can't think of what I would want a list for - as I'd have to load a different design if swapping between a different number of frames per step.

Reply:
I think you'd only want a divide into 3 if you had 3 detectors, and I hope we never have a beamline with 3 beams...

Reply:
I guess K11 has "three" beams but 2 (pink and mono) should only be used one at a time.

Should I modify this to only handle a fixed value of 2 and rename the part to DoubleTriggerPart? DoubleFramesPerStepPart?

I named it a more generic term in the first place because of EtherCAT also wanting these doubling triggers.

Reply:
That makes sense

Reply:
The part is now called DoubleTriggerPart and checks that frames_per_step is always equal to 2.